core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -681,7 +694,9 @@ private void performLocalAction(ClusterState state, ShardRouting primary, Discov
                 logger.trace("send action [{}] on primary [{}] for request [{}] with cluster state version [{}] to [{}] ",
                     transportPrimaryAction, request.shardId(), request, state.version(), primary.currentNodeId());
             }
-            performAction(node, transportPrimaryAction, true, new ConcreteShardRequest<>(request, primary.allocationId().getId()));
+            final long primaryTerm = state.metaData().getIndexSafe(primary.index()).primaryTerm(primary.id());
```

Comment:
nit: we can pass the indexMetaData (or primary term) as an argument. We look it up on in the calling method. Save on double lookups.